services/variableResolver.js

```diff
@@ -0,0 +1,57 @@
+import _ from 'lodash';
+import autoBind from 'auto-bind';
+
+const THEME_EXTENSION_SCOPE = 'shoutem.theme';
+
+export class ThemeVariableResolver {
+  constructor(variables) {
+    this.variables = { ...variables };
+
+    autoBind(this);
+  }
+
+  setVariables(variables) {
+    this.variables = { ...variables };
+  }
+
+  getScopedVariable(scope, variable) {
+    if (!this.variables || _.isEmpty(this.variables)) {
+      throw new Error(
+        'Theme variable resolver not yet initialised. Please call `setVariables` prior to this function call',
+      );
+    }
+
+    const variablePath = _.split(variable, '.');
+
+    return _.get(
+      this.variables,
+      [scope, ...variablePath],
+      _.get(this.variables, variablePath),
+    );
+  }
+
+  // Currently supports 2 function signture
+  // resolveVariable(variableName) -> Tries to get the variable under the shoutem.theme namespace
+  // resolveVariable(scope, variableName) -> Tries to get the variable under specific scope
+  // Both signatures default to root parameter for the required variable name
+  // Variable name can be stringified object path or a regular string, i.e -> text.color
+  resolveVariable(...params) {
+    if (arguments.length === 1) {
```

Comment:
`params` instead of `arguments`?

Or am I missing something?

Reply:
Also, I'm not certain I understand why arguments are passed as `...params` or why `params.length` is used, would it not be better to receive two arguments if we only expect two arguments? What am I missing?

Reply:
lint complains about arguments syntax, so that's why the spread.

Reasoning behind is the fact that we want to retain simple usage for UI users outside shoutem. For them, there are no scopes unless they want them to exist by defining custom variables. So, the method signature remains simple resolveVariable(variableName).

Other use case is shoutem development scenario, where we target the scoped variable, so we use resolveVariable(scope,variableName)

In both cases, we default to variables contained in the root of the vars object. This is just way less confusing for people using the standalone package.

Another argument is less abstraction for the custom devs. For them, there are general variables ( theme variables ), and scoped ones. Under the hood, general variables are also scoped, but there's no reason for the dev to be aware of that ( simpler interface ).

let me know if i make sense :D

Reply:
Why not use `resolveVariable(variableName, scope)` then? If `scope` exists, you apply it, otherwise you just use `resolveVariable(variableName)`.

I don't understand what eslint would complain about.

Sorry, something probably just isn't clicking in my head regarding this, I just don't see the need for using the param spread or checking length of params

Reply:
linter explicitly told me to use spread instead of arguments syntax... He doesn't like it i guess.

What you suggesting was my initial approach, but i wanted the scope to go first. It's more natural somehow to me( probably because we follow the actual object structure ). 

`resolveVariable('shoutem.navigation', 'backgroundColor')`. We can change it to what you're suggesting if this doesn't look better to you guys

Reply:
I see what you mean, I think it's more important to keep the scope first than it is to have one function defined without spread params for the sake of not having spread params.